packages/cli/generators/datasource/index.js

```diff
@@ -0,0 +1,290 @@
+      connectorChoices.push({
+        name: `${connector.description} ${chalk.gray(support)}`,
+        value: connector.name,
+      });
+    });
+
+    this.connectorChoices = connectorChoices;
+    // Add `other` so users can add a connector that isn't part of the list
+    // Though it can be added by creating a PR and adding it to
+    // connectors.json
+    this.connectorChoices.push('other');
+
+    return super._setupGenerator();
+  }
+
+  /**
+   * Ensure CLI is being run in a LoopBack 4 project.
+   */
+  checkLoopBackProject() {
+    return super.checkLoopBackProject();
+  }
+
+  /**
+   * Ask for DataSource Name -- Must be unique
+   */
+  promptArtifactName() {
+    return super.promptArtifactName();
+  }
+
+  /**
+   * Ask the user to select the connector for the DataSource
+   */
+  promptConnector() {
+    debug('prompting for datasource connector');
+    const prompts = [
+      {
+        name: 'connector',
+        message: `Select the connector for ${chalk.yellow(
+          this.artifactInfo.name,
+        )}:`,
+        type: 'list',
+        default: 'memory',
+        choices: this.connectorChoices,
+      },
+    ];
+
+    return this.prompt(prompts).then(props => {
+      Object.assign(this.artifactInfo, props);
+      return props;
+    });
+  }
+
+  /**
+   * If the user selected `other` for connector -- ask the user to provide
+   * `npm` module name for the connector.
+   */
+  promptCustomConnectorInfo() {
+    if (this.artifactInfo.connector !== 'other') {
+      debug('custom connector option was not selected');
+      return;
+    } else {
+      debug('prompting for custom connector');
+      const prompts = [
+        {
+          name: 'customConnector',
+          message: "Enter the connector's module name",
+          validate: utils.validate,
+        },
+      ];
+
+      return this.prompt(prompts).then(props => {
+        this.artifactInfo.connector = props.customConnector;
+        return props;
+      });
+    }
+  }
+
+  /**
+   * Prompt the user for connector specific settings -- only applies to
+   * connectors in the connectors.json list
+   */
+  promptConnectorConfig() {
+    debug('prompting for connector config');
+    // Check to make sure connector is from connectors list (not custom)
+    const settings = connectors[this.artifactInfo.connector]
+      ? connectors[this.artifactInfo.connector]['settings']
+      : {};
+
+    const prompts = [];
+    // Create list of questions to prompt the user
+    Object.entries(settings).forEach(([key, setting]) => {
+      // Set defaults and merge with `setting` to override properties
+      const question = Object.assign(
+        {},
+        {name: key, message: key, suffix: ':'},
+        setting,
+      );
+
+      /**
+       * Allowed Types: string, number, password, object, array, boolean
+       * Must be converted to inquirer types -- input, confirm, password
+       */
+      switch ((setting.type || '').toLowerCase()) {
```

Comment:
It might be better style to determine `question.type` in an intermediary variable and then assign it all together in line 141 (using `Object.assign`)